### PR TITLE
Proposed fix for issue #1: autofix option

### PR DESCRIPTION
**Pull request by ericqweinstein**
Initial stab at a fix for #1. Comments/discussion welcome!

### CODE REVIEW

File: lib/ruumba/analyzer.rb [hunk 1]
```diff
@@ -61,12 +61,50 @@ def copy_erb_files(fq_dir, tmp, pwd)
...
+            out << if line.match?(/<%/)
+                     "  <% #{code.shift.rstrip} %>\n"
+                   elsif /<%=/
+                     "  <%= #{code.shift.rstrip} %>\n"
+                   else
+                     noncode.shift
+                   end
+          end
+
+          File.open("#{src_dir}/#{src}", 'w') do |contents|
```
Comment (ericqweinstein) [inline lib/ruumba/analyzer.rb]: Should probably DRY this up a bit

**Comment by ericqweinstein** [comment]
I also noted in the README that this feature is experimental, as I haven't extensively tested it.

**Comment by SijiaDavis** [comment]
Hi @ericqweinstein, I tested it on my project and I don't think the ruby code embedded in the `.erb` files are corrected... I think they are corrected in the `.erb.rb ` files, but the changes are not reflected in the original `.erb` file.

**Comment by ericqweinstein** [comment]
@SijiaDavis Here are the steps I'm using to repro; could you confirm whether this does/does not work for you?

* Clone the PR branch
* Create an example.erb in the root of the directory:

```erb
<p>
  <% test = true; puts 'Hello!' if !test %>
</p>
```

* Run the following:

```bash
$ gem build ruumba.gemspec && gem install ruumba-0.1.2.gem
$ ruumba --auto-correct .
```

When you open example.erb, you should now see:

```erb
<p>
  <% test = true; puts 'Hello!' unless test %>
</p>
```

File: lib/ruumba/analyzer.rb [hunk 2]
```diff
@@ -61,12 +61,50 @@ def copy_erb_files(fq_dir, tmp, pwd)
...
+          noncode    = File.readlines("#{src_dir}/#{src}")
+                           .reject { |l| l =~ ERB_REGEX }
+          to_correct = File.readlines("#{src_dir}/#{src}")
+          out        = []
+
+          to_correct.each do |line|
+            # @TODO: This is pretty hacky. (EQW 30 Aug 2017)
+            out << if line.match?(/<%/)
+                     "  <% #{code.shift.rstrip} %>\n"
+                   elsif /<%=/
```
Comment (ericqweinstein) [inline lib/ruumba/analyzer.rb]: This should be:

```rb
elsif line.match?(/<%=/)
```

File: lib/ruumba/analyzer.rb [hunk 3]
```diff
@@ -33,8 +33,8 @@ def run(dir = ARGV)
     # @return [String] The extracted Ruby code.
     def extract(filename)
       File.read(filename).scan(ERB_REGEX).map(&:last)
-        .reject { |line| line[0] == '#' }
-        .map(&:strip).join("\n")
+          .reject { |line| line[0] == '#' }
```
Comment (ericqweinstein) [inline lib/ruumba/analyzer.rb]: Not sure why RuboCop bumped the indentation here

File: lib/ruumba/analyzer.rb (hunk 2 above)
Comment (ericqweinstein) [reply lib/ruumba/analyzer.rb]: @SijiaDavis This could be the source of your trouble; I'll push up a correction tonight or tomorrow.

**Comment by SijiaDavis** [comment]
Hi @ericqweinstein, I followed your steps in your last comment and it worked indeed. However, it's not working for me if I use the ruumba gem to auto-correct erb contents in other projects.  

What I did:
- Clone the PR branch
- Build and install the gem 
- cd into another ruby on rails project
- Create the `.ruumba.yml` file (follow the instruction in the README file)
- `$ ruumba -D --auto-correct app/views -c .ruumba.yml`
- Check the files that are marked as 'corrected' according to the log displayed in the terminal, they are not changed however...

Please let me know if I'm doing something wrong?

**Comment by SijiaDavis** [comment]
Hi @ericqweinstein, I followed your step in your last comment and it worked (fixed the file in the same directory), however, it's not working for me when I try to use the ruumba gem in my other projects.

What I did:
- Clone the PR branch
- Build and Install the gem
- cd into another rails project
- Create the `.ruumba.yml` file in the root directory (follow the instruction in the README)
- `$ ruumba --auto-correct app/views -c .ruumba.yml`
- Check lines that are marked as "Corrected" in the terminal log, but they are actually not corrected...

Let me know if I am missing something, thanks!

File: lib/ruumba/analyzer.rb (hunk 2 above)
Comment (SijiaDavis) [reply lib/ruumba/analyzer.rb]: Nice catch 👍🏽, unfortunately I don't think this is the source of why it wasn't working for me... I have some `<% ... %>` in my erb files that were supposed to be corrected but weren't...

File: lib/ruumba/analyzer.rb (hunk 3 above)
Comment (SijiaDavis) [reply lib/ruumba/analyzer.rb]: probably to align with `.read`

**Comment by ericqweinstein** [comment]
@SijiaDavis There might have been an interpolation bug; I pushed up a fix for it. Could you give it another try?

File: lib/ruumba/analyzer.rb [hunk 4]
```diff
@@ -61,12 +61,50 @@ def copy_erb_files(fq_dir, tmp, pwd)
...
+      end
+    end
+
     def run_rubocop(target, tmp)
       args = (@options[:arguments] || []).join(' ')
       todo = tmp + '.rubocop_todo.yml'
 
       system("cd #{tmp} && rubocop #{args} #{target}").tap do
         FileUtils.cp(todo, ENV['PWD']) if File.exist?(todo)
+        autofix!(target, tmp) if @options[:arguments]&.include? '--auto-correct'
```
Comment (ceritium) [inline lib/ruumba/analyzer.rb]: This is not compatible with old versions of ruby :(

Could be this implemented without the `&.` operator?

**Comment by ceritium** [comment]
@ericqweinstein I am having the same issue that @SijiaDavis

**Comment by ericqweinstein** [comment]
@ceritium @SijiaDavis Got it. Will take a look early next week (apologies, out of town for a wedding this weekend)

File: lib/ruumba/analyzer.rb (hunk 4 above)
Comment (ericqweinstein) [reply lib/ruumba/analyzer.rb]: Sure, I can downgrade this branch back to 2.2.2+ rather than 2.3.0+

**Comment by ericqweinstein** [comment]
@ceritium @SijiaDavis Apologies, between work and school, I'm a bit swamped this week. Please feel free to pick this up and run with it if you like; otherwise, I'll get to it as soon as I can 👍

**Pull request closed without merging**